modules/core/react/lib/DirectionProvider.tsx

```diff
@@ -0,0 +1,16 @@
+import rtlCSSJS from 'rtl-css-js';
+import {CSSObject} from '@emotion/css';
+
+export enum Direction {
+  LTR,
+  RTL,
+}
+
+function DirectionProvider(dir: Direction, styles: CSSObject) {
```

Comment:
I'm not sure we should call this a `Provider`, open to suggestions

Reply:
You could mimic the `useTheme` hook and end up wtih something like:

```tsx
styled('div')(
  useDirection(direction)({
    //styles
  })
)
```

maybe?

Reply:
However, this doesn't account for an overriding `isRTL` prop (or something similar).

Reply:
Why wouldn't it? direction could be set at the component level, where we look at HTML, Context and then prop?